packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -75,6 +78,23 @@ export default class EthereumAnchorService extends AnchorService {
         this._ceramic = ceramic
     }
 
+    async init(): Promise<void> {
+        // Get the chainIds supported by our anchor service
+        const response = await fetch(this.chainIdApiEndpoint)
+        const json = await response.json()
+        this._supportedChains = json.supportedChains
+
+        // Confirm that we have an eth provider that works for each of the chains that the anchor service supports
+        for (const chain of this._supportedChains) {
```

Comment:
Hmm trying to understand what is happening here. We have supported chains from the perspective of the anchor service, but also only 2 supported chains listed here eip155:1 and eip155:3.

Should we just be checking that the anchor service supports __*at least* one of these two__ chains?
Seems like this loop should not be needed. Whichever chain is in our supported list is the one we can use to get the provider.

And then, in the case that they pass an rpc endpoint, we also ensure it is one of the two chains.

Maybe I'm missing something but not sure entirely what we are trying to achieve here yet.

Reply:
So we do support chainids other than `eip155:1` and `eip155:3` (for instance ganache uses chainid `eip155:1337`).  But to use the default ethers provider we need the network name not the chainId, so these are the two hardcoded mappings from chainid to network name.  We should probably add Rinkeby to this list.

The loop probably isn't necessary at the moment, but is future proofing in case the CAS supports multiple chains in the future.  In that world, we'd need to make sure that the ceramic node had an eth provider for every chain that the CAS supports.  Right now the CAS will never return more than one supported chainId at a time, so it doesn't really matter either way.

Reply:
So idea is to allow anchoring to multiple chains at once?

Reply:
Also not sure it’s fully future proof it seems because we only allow you to pass 1 rpc endpoint which I think can only have 1 network at a time. I could be wrong..

Reply:
Interesting point @valmack! Maybe we need to allow passing multiple endpoints for specific chainIds?

Reply:
Yeah, we don't actually support multiple chains yet, there would be more work to do to make that work well.  We just made the API from the CAS for fetching the supported chain(s) use an array so that we could add this support in the future without breaking the API.  I added an assert that there is only one supported chain returned from the CAS for now, which makes this more explicit.